runtime/drivers/clickhouse/olap.go

```diff
@@ -224,8 +225,20 @@ func (c *connection) CreateTableAsSelect(ctx context.Context, name string, view
 	create.WriteString("CREATE OR REPLACE TABLE ")
 	create.WriteString(safeSQLName(name))
 
-	// see if there is any column override
-	if outputProps.Columns != "" {
+	if outputProps.Columns == "" {
+		// infer columns
+		v := tempName("view")
+		err := c.Exec(ctx, &drivers.Statement{Query: fmt.Sprintf("CREATE OR REPLACE VIEW %s AS %s", v, sql)})
+		if err != nil {
+			return err
+		}
+		defer func() {
+			_ = c.Exec(context.Background(), &drivers.Statement{Query: "DROP VIEW " + v})
+		}()
```

Comment:
Can we use a ctx with some kind of default timeout here?

Reply:
Sure. We can keep a 5 second timeout.